src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -101,7 +101,7 @@
   /**
    * A cache of local repositories.
    */
-  final LRUMap<GitHubProject, LocalRepository> localRepositories
+  static final LRUMap<GitHubProject, LocalRepository> localRepositories
```

Comment:
`LRUMap` is not thread-safe.

https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/map/LRUMap.html

Please make sure that there are no race conditions when the cache is accessed by multiple threads.

Reply:
I have moved the variable usage inside the `synchronized` block to be thread-safe.